packages/ra-ui-materialui/src/input/AutocompleteInput.tsx

```diff
@@ -192,6 +191,19 @@ export const AutocompleteInput = <
 
     const translate = useTranslate();
 
+    const finalChoices =
+        emptyText == null || isRequiredOverride
+            ? allChoices
+            : [
+                  {
+                      [optionValue || 'id']: '',
```

Comment:
My question is if I should add back `emptyValue` and use it here:

```
                      [optionValue || 'id']: emptyValue,
```

Reply:
If it's cheap, yes

Reply:
but but but... if the user provides an `emptyValue` that is not a string, this won't work. So we should document that limitation and throw an error if the users passes a non-string `emptyValue`.